### DISCUSSION TITLE
Usage of "to-be-defined" variables from parameters in usecase

### DISCUSSION THREAD

**Discussion by AnatolManikalo** [SAP-archive/btp-setup-automator#449] [Q&A]
Hello colleagues,

Is there a possibility to use variables from the parameters.json file in usecase.json, if these variables' values are only created during the subaccount creation?

Example: "subaccountid" and "orgid" values are not declared in the initial parameters files, because these values are generated during the subaccount creation and are not known beforehand. As an account owner, I want to create labels for newly created subaccount which will include CF Org ID of the subaccount. To achieve this I want to use the following "executeAfterAccountSetup" command: 

`{
      "description": "Set Labels",
      "command": "btp update accounts/subaccount $subaccountid --labels \"{\"cf-org\":[\"$orgid\"]}\""
    }`

Is there a way to achieve this?

Thanks,
ANatol

**Comment by lechnerc77** [comment]
Hi Anatol,

this is only possible via actions defined  in the `executeAfterAccountSetup` section.

Best,
Christian

**Reply by AnatolManikalo** [reply]
Hi Christian,
Thanks for quick reply. 

That's exactly how I'm trying to do this, but I'm getting the error message and can see that variable's value is not inserted:

![image](https://user-images.githubusercontent.com/32706007/222709968-7c7e00a6-2bd2-4635-99f1-9d56c20bfb3c.png)

Am I doing something wrong?

Usecase extract for this:

```
{
      "description": "Set Labels",
      "command": "btp update accounts/subaccount $subaccountid --labels \"{\"cf-org\":[\"$orgid\"],\"idm-arm\":[\"N/A\"],\"mfa-enabled\":[\"true\"],\"type\":[\"playground\"]}\""
    }
```

**Reply by AnatolManikalo** [reply]
If I run the following command manually — it works:

`btp update accounts/subaccount c8b64198-94f5-4509-aea9-7572bd8d2f2e --labels "{\"cf-org\":[\"orgid\"],\"idm-arm\":[\"N/A\"],\"mfa-enabled\":[\"true\"],\"type\":[\"playground\"]}"`

So looks like there are 2 things:
1) $subaccountid and $orgid variables are not called 
2) BTPSA removes \ symbols to escape quotation marks and as a result btp cli doesn't accept json format

Could you please advise here?

**Reply by lechnerc77** [reply]
Ad 1) You need to fetch them from the  /home/user/log/metadata_log.json Should be something like:

- SUBACCOUNTID=$(jq -r '."subaccountid"' /home/user/log/metadata_log.json)
- ORG=$(jq -r '."org"' /home/user/log/metadata_log.json)

Ad 2) tbh, would need to check if we strip anything here (did not see anything at a first glance), but not sure when I can take a look.

**Comment by rui1610** [comment]
The only way to do this, is via the way that @lechnerc77 explained, by fetching the information in the /home/user/log/metadata_log.json file e.g. via jq and injecting in your code through the executeAfterAccountSetup section.
